contracts/Vault.sol

```diff
@@ -833,28 +829,32 @@ contract Vault is
     /**
      * Validates conditions for unsponsoring amount @param _amount of the deposit with the id @param _id.
      *
-     * @notice The NFTs of the deposit will be burned if the deposited amount is equal to the amount being withdrawn.
-     *
-     * @param _id Id of the deposit.
+     * @param _to Address that will receive the funds.
+     * @param _tokenId Id of the deposit.
      * @param _amount Amount to be unsponsored/withdrawn.
      */
-    function _unsponsorSingle(uint256 _id, uint256 _amount) internal {
-        Deposit memory _deposit = deposits[_id];
+    function _unsponsorSingle(
+        address _to,
+        uint256 _tokenId,
+        uint256 _amount
+    ) internal {
+        Deposit memory _deposit = deposits[_tokenId];
 
         if (_deposit.owner != msg.sender) revert VaultNotAllowed();
         if (_deposit.lockedUntil > block.timestamp) revert VaultAmountLocked();
         if (_deposit.claimerId != address(0)) revert VaultNotSponsor();
         if (_deposit.amount < _amount)
             revert VaultCannotWithdrawMoreThanAvailable();
 
-        if (_amount != _deposit.amount) {
-            deposits[_id].amount -= _amount;
-            return;
-        }
+        bool isFull = _amount == _deposit.amount;
 
-        delete deposits[_id];
+        if (!isFull) {
+            deposits[_tokenId].amount -= _amount;
+        } else {
+            delete deposits[_tokenId];
+        }
 
-        emit Unsponsored(_id);
+        emit Unsponsored(_tokenId, _amount, _to, isFull);
```

Comment:
I think we also need to make changes to the subgraph

Reply:
thanks, please do so :pray:

Reply:
imo would be nice to use if () {} return; instead of if / else here

you can emit the event before the if return stuff if needed, sequence should not matter

Reply:
I must notice you really don't like `else` statements :slightly_smiling_face: I also don't like to use them but in this particular situation, the code looks more readable with `else` in my opinion

Reply:
I disagree to be honest, two branches are always harder two read than one imo

And yes I really really hate them

Reply:
you guys are funny xD

Reply:
:pray: